test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,88 @@
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  - interval: 1m
+    input_series:
+
+      # Pod is Pending, not unschedulable and its alerted
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Pending"}'
+        values: '0x15'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-1", pod="pod-1"}'
+        values: '0x15'
+
+      # Pod is Pending, unschedulable and its alerted
+      - series: 'kube_pod_status_phase{pod="pod-2", namespace="ns-1", phase="Pending"}'
+        values: '0x15'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-1", pod="pod-2"}'
+        values: '1x10 0x5'
+
+      # Pod is Pending with value 1, not unschedulable and its alerted
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-2", phase="Pending"}'
+        values: '1x15'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-2", pod="pod-1"}'
+        values: '0x15'
+
+      # Pod is Unknown, unschedulable and its not alerted
+      - series: 'kube_pod_status_phase{pod="pod-3", namespace="ns-2", phase="Unknown"}'
+        values: '0x15'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-2", pod="pod-3"}'
+        values: '1x15'
+
+      # Pod is Running, not unschedulable and its not alerted
+      - series: 'kube_pod_status_phase{pod="pod-3", namespace="ns-1", phase="Running"}'
+        values: '1x10 0x5'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-1", pod="pod-3"}'
+        values: '0x15'
+
+      # Pod is Pending, schedulable and its alerted
```

Comment:
This is also not alerted and pod is unscheduled for most of the time and only available for some window in between.

Reply:
Updated